### PR TITLE
EVA-1283 Store separately RS that were declustered

### PR DESCRIPTION
**Pull request by andresfsilva**

### CODE REVIEW

File: eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriter.java [hunk 1]
```diff
@@ -24,9 +24,7 @@
 import uk.ac.ebi.eva.accession.dbsnp.persistence.DbsnpClusteredVariantEntity;
 import uk.ac.ebi.eva.accession.dbsnp.persistence.DbsnpVariantsWrapper;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
+import java.util.*;
```
Comment (cyenyxe) [inline eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriter.java]: According to our guidelines it should be one import per class.

File: eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriter.java [hunk 2]
```diff
@@ -77,13 +75,7 @@ private void writeClusteredVariants(List<? extends DbsnpVariantsWrapper> items)
 
     private void writeClusteredVariantsDeclustered(List<DbsnpClusteredVariantEntity> clusteredVariantsDeclustered) {
         if (!clusteredVariantsDeclustered.isEmpty()) {
```
Comment (cyenyxe) [inline eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriter.java]: You don't need to do this, the writer takes care of empty sets.

File: eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantDeclusteredWriterTest.java [hunk 3]
```diff
@@ -64,6 +80,11 @@ public void writeDeclusteredRs() {
                                                                               clusteredVariant);
         writer.write(Collections.singletonList(variant));
 
+        List<DbsnpClusteredVariantEntity> rsEntities = mongoTemplate.find
+                (new Query(), DbsnpClusteredVariantEntity.class, DBSNP_CLUSTERED_VARIANT_DECLUSTERED_COLLECTION_NAME);
+
+        assertEquals(1, rsEntities.size());
+        assertEquals(variant, rsEntities.get(0));
     }
```
Comment (cyenyxe) [inline eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantDeclusteredWriterTest.java]: Sorry I didn't mention this in the previous review. Could you add a test to check insertions of multiple distinct entities, and another one with duplicates?

File: eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriter.java (hunk 2 above)
Comment (cyenyxe) [reply eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriter.java]: We found out that it doesn't 😞

File: eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java [hunk 4]
```diff
@@ -225,6 +233,13 @@ private void assertDeclusteringHistoryStored(DbsnpVariantsWrapper wrapper) {
         assertEquals(1, importCounts.getOperationsWritten());
     }
 
+    private void assertClusterVariantDeclusteredStored(DbsnpVariantsWrapper wrapper) {
+        List<DbsnpClusteredVariantEntity> rsDeclusteredEntities = mongoTemplate.find
+                (new Query(), DbsnpClusteredVariantEntity.class, "dbsnpClusteredVariantEntityDeclustered");
```
Comment (pabarcgar) [inline eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java]: Use constant instead of hardcoded string

File: eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java [hunk 5]
```diff
@@ -234,17 +249,54 @@ public void repeatedClusteredVariants() throws Exception {
...
+        wrapper2.setOperations(Collections.singletonList(operationEntity2));
+
+        dbsnpVariantsWriter.write(Arrays.asList(wrapper1, wrapper2));
+        assertClusteredVariantStored(wrapper1);
+        assertClusteredVariantDeclusteredStored(wrapper1);
+    }
 
-        dbsnpVariantsWriter.write(Arrays.asList(firstWrapper, secondWrapper));
-        assertClusteredVariantStored(firstWrapper);
+    private void assertClusteredVariantDeclusteredStored(DbsnpVariantsWrapper wrapper) {
```
Comment (pabarcgar) [inline eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java]: what is the difference between this method and this other one?
assertClusterVariantDeclusteredStored

Comment (andresfsilva) [reply eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java]: It was the same, deleted this method
